block/retriever.go

```diff
@@ -37,26 +40,36 @@ func (m *Manager) RetriveLoop(ctx context.Context) {
 // syncUntilTarget syncs the block until the syncTarget is reached.
 // It fetches the batches from the settlement, gets the DA height and gets
 // the actual blocks from the DA.
-func (m *Manager) syncUntilTarget(ctx context.Context, syncTarget uint64) {
+func (m *Manager) syncUntilTarget(ctx context.Context, syncTarget uint64) error {
 	currentHeight := m.store.Height()
 	for currentHeight < syncTarget {
-		m.logger.Info("Syncing until target", "current height", currentHeight, "syncTarget", syncTarget)
-		resultRetrieveBatch, err := m.settlementClient.RetrieveBatch(atomic.LoadUint64(&m.lastState.SLStateIndex) + 1)
+		currStateIdx := atomic.LoadUint64(&m.lastState.SLStateIndex) + 1
+		m.logger.Info("Syncing until target", "height", currentHeight, "state_index", currStateIdx, "syncTarget", syncTarget)
+		settlementBatch, err := m.settlementClient.RetrieveBatch(currStateIdx)
```

Comment:
The retry before was done using the for loop - i.e while we couldn't sync until target we keep trying. now on the first error you panic (e.g if the rpc of the hub was down). 
It's fine in essence to panic once an error is reached, but I think we should add a retry mechanism inside the retreiveBatch logic (specifically in the dymension client) as we don't want to panic on the first attempt.

Reply:
will fix. in the original the retry would happen on the next state update event?
as the `for loop` will be blocked on `syncTarget := syncTargetpoller.Next()`